src/content/contribute/plugin-patterns.md

```diff
@@ -10,63 +12,66 @@ Plugins grant unlimited opportunity to perform customizations within the webpack
 After a compilation is sealed, all structures within the compilation may be traversed.
 
 ```javascript
-function MyPlugin() {}
-
-MyPlugin.prototype.apply = function(compiler) {
-  compiler.plugin('emit', function(compilation, callback) {
-
-    // Explore each chunk (build output):
-    compilation.chunks.forEach(function(chunk) {
-      // Explore each module within the chunk (built inputs):
-      chunk.modules.forEach(function(module) {
-        // Explore each source file path that was included into the module:
-        module.fileDependencies.forEach(function(filepath) {
-          // we've learned a lot about the source structure now...
+class MyPlugin {
+  apply(compiler) {
+    compiler.hooks.emit.tapAsync("MyPlugin", (compilation, callback) => {
```

Comment:
we use single quotes for \`\`\`javascript i wont mark all of the places where you changed it or added as `"`, please change everywhere